### PR TITLE
Story/quad 62

### PR DESCRIPTION
**Pull request by ajaymodi**
MERGED STORY/QUAD-63 TO QUAD-62 AND INTEGRATED SUGGESTIONS.

### CODE REVIEW

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java [hunk 1]
```diff
@@ -154,10 +173,63 @@
...
+                        ja.put(j);
+        
+                    } catch (JSONException e) {
+                        logger.error("JSONException:", e);
+                        return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+                }
+            }
+            
+        }catch (QuadrigaStorageException e) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java:203]: this catch block should be where the exception is actually thrown (line 200 I believe).

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java [hunk 2]
```diff
@@ -448,5 +519,58 @@
...
+                        logger.error("JSONException:", e);
+                        return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+
+                }
+            }
+            
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+            
+        } catch (QuadrigaStorageException e) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java]: same here. put the catch block where it is thrown.

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java [hunk 3]
```diff
@@ -185,6 +205,55 @@
...
+                        
+                }
+                
+            }
+        }
+        catch(JSONException e){
+            logger.error("JSONException:", e);
+            return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        catch(QuadrigaStorageException e){
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:233]: same here

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java [hunk 4]
```diff
@@ -404,5 +474,57 @@
...
+                        // TODO Auto-generated catch block
+                        logger.error("JSONException:", e);
+                        return new ResponseEntity<String>(e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+                    }
+                }
+            }
+                 
+            return new ResponseEntity<String>(ja.toString(), HttpStatus.OK);
+
+        } catch (QuadrigaStorageException e) {
```
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/DictionaryRestController.java:506]: same here

File: Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspace.jsp [hunk 5]
```diff
@@ -32,28 +50,50 @@
...
+			<li data-jstree='{"icon":"/quadriga/resources/txt-layout/css/images/pen.png"}'><a href="${pageContext.servletContext.contextPath}/auth/workbench/${project.projectId}/updatecollaborators">
+			Update</a></li>
+		</ul>
 		</li>
-		
+		 -->
 	</ul>
 </div>
+<h5 class="major" style="margin-top: 0.5em;margin-bottom: 0em;">
+	<span>DICTIONARIES</span>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspace.jsp]: should be "Dictionaries" (not all caps)

File: Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspace.jsp [hunk 6]
```diff
@@ -32,28 +50,50 @@
...
+</h5>
+<div id="projectDictionaries">
+	<a href="${pageContext.servletContext.contextPath}/auth/workbench/${project.projectId}/adddictionary"><i class="fa fa-plus-circle"></i> Add</a>
+	<a href="${pageContext.servletContext.contextPath}/auth/workbench/${project.projectId}/deletedictionary"><i class="fa fa-times-circle"></i> Delete</a>
+</div>
+<h5 class="major" style="margin-top: 0.5em;margin-bottom: 0em;">
+	<span>CONCEPT COLLECTIONS</span>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspace.jsp]: Should be "Concept Collections"

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java (hunk 2 above)
Comment (ajaymodi) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java]: According to eclipse, it wants to surround object with try catch when it uses put method. So I believe this exception is thrown when we try to add put method on object.

File: Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspacecontent.jsp [hunk 7]
```diff
@@ -5,6 +5,8 @@
 	href="${pageContext.servletContext.contextPath}/resources/txt-layout/css/style.min.css" />
 <script
 	src="${pageContext.servletContext.contextPath}/resources/txt-layout/js/jstree.min.js"></script>
+<link
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspacecontent.jsp]: this should be in the skeleton not in a specific jsp page

File: Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspacecontent.jsp [hunk 8]
```diff
@@ -19,55 +21,39 @@
...
+
+<h5 class="major" style="margin-top: 0.5em;margin-bottom: 0em;">
+	<span>DICTIONARIES</span>
+</h5>
+<div id="workspaceDictionaries">
+	<a href="${pageContext.servletContext.contextPath}/auth/workbench/workspace/${workspacedetails.workspaceId}/adddictionary"><i class="fa fa-plus-circle"></i> Add</a>
+	<a href="${pageContext.servletContext.contextPath}/auth/workbench/workspace/${workspacedetails.workspaceId}/deletedictionary"><i class="fa fa-times-circle"></i> Delete</a>
+</div>
+<h5 class="major" style="margin-top: 0.5em;margin-bottom: 0em;">
+	<span>CONCEPT COLLECTIONS</span>
```
Comment (jdamerow) [inline Quadriga/src/main/webapp/WEB-INF/views/navigation/nav-workspacecontent.jsp]: same as before (no caps)

File: Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java (hunk 2 above)
Comment (jdamerow) [inline Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java]: you mean: 

```
j.put("id", conceptCollection.getConceptCollection() .getConceptCollectionId());
j.put("name", conceptCollection.getConceptCollection().getConceptCollectionName());
```

? 
Yes, the json exception is fine where it is, but the Quadriga exception should be thrown before.

**Comment by ajaymodi** [comment]
Please check now. Incorporated changes.

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
retest this please

**Comment by jdamerow** [comment]
The lists of dictionaries/concept collections should have the icons that concept collections and dictionaries have.
image
=>
image
image

**Pull request merged**